kratos/solving_strategies/strategies/line_search_strategy.h

```diff
@@ -133,21 +133,60 @@ class LineSearchStrategy
         bool CalculateReactions = false,
         bool ReformDofSetAtEachStep = false,
         bool MoveMeshFlag = false
-    ): ResidualBasedNewtonRaphsonStrategy<TSparseSpace, TDenseSpace, TLinearSolver>(model_part, pScheme, pNewLinearSolver,pNewConvergenceCriteria,MaxIterations,CalculateReactions,ReformDofSetAtEachStep, MoveMeshFlag)
-    {}
+    ): ResidualBasedNewtonRaphsonStrategy<TSparseSpace, TDenseSpace, TLinearSolver>(model_part, pScheme, pNewLinearSolver,pNewConvergenceCriteria,MaxIterations,CalculateReactions,ReformDofSetAtEachStep, MoveMeshFlag),
+       mSettings(Parameters(R"({})"))
+    {
+        CheckDefaultsAndAssignSettings();
+    }
+
+    LineSearchStrategy(
+        ModelPart& model_part,
+        typename TSchemeType::Pointer pScheme,
+        typename TLinearSolver::Pointer pNewLinearSolver,
+        typename TConvergenceCriteriaType::Pointer pNewConvergenceCriteria,
+        typename TBuilderAndSolverType::Pointer pNewBuilderAndSolver,
+        int MaxIterations = 30,
+        bool CalculateReactions = false,
+        bool ReformDofSetAtEachStep = false,
+        bool MoveMeshFlag = false
+    ): ResidualBasedNewtonRaphsonStrategy<TSparseSpace, TDenseSpace, TLinearSolver>(model_part, pScheme, pNewLinearSolver,pNewConvergenceCriteria,pNewBuilderAndSolver,MaxIterations,CalculateReactions,ReformDofSetAtEachStep, MoveMeshFlag),
+       mSettings(Parameters(R"({})"))
+    {
+        CheckDefaultsAndAssignSettings();
+    }
+
+    LineSearchStrategy(
+        ModelPart& model_part,
+        typename TSchemeType::Pointer pScheme,
+        typename TLinearSolver::Pointer pNewLinearSolver,
+        typename TConvergenceCriteriaType::Pointer pNewConvergenceCriteria,
+        Parameters Settings,
+        int MaxIterations = 30,
```

Comment:
You don't need the other inputs, can be all replaced by the Parameters

Reply:
Follow the name of the parameters used as input on the python solvers

Reply:
thing is that this parameter goes to Newton Raphson strategy so I cannot remove it

Reply:
> Follow the name of the parameters used as input on the python solvers

it uses settings...

Reply:
We can add a Parameters constructor to the NR, then you can. This doesn't break previous behavior, and I think everyone will agree with that

Reply:
Mmm problem is then you have you validate the settings in the base class as there will be settings that will belong to the derived class